src/main/java/com/splunk/hecclient/HecAckPoller.java

```diff
@@ -184,6 +184,41 @@ public int getAckPollInterval() {
         return ackPollInterval;
     }
 
+    public void stickySessionHandler(HecChannel channel) {
+        // Set channel unavailable
+        channel.setAvailable(false);
+        log.debug("Channel {} set to be not available", channel.getId());
+        // Get batches for the channel
+        ConcurrentHashMap<Long, EventBatch> channelBatches = outstandingEventBatches.get(channel);
+        // Remove batches for the channel from the poller
+        if(channelBatches != null && channelBatches.size() > 0) {
+            totalOutstandingEventBatches.addAndGet(-channelBatches.size());
```

Comment:
Possibly use getAndDecrement() after each batch.fail instead of larger subtraction of totalOutstandingEventBatches to ensure correctness in value at all times.